docs/csharp/programming-guide/concepts/async/index.md

```diff
@@ -24,7 +24,10 @@ For a parallel algorithm, you'd need multiple cooks (or threads). One would make
 
 Now, consider those same instructions written as C# statements:
 
-[!code-csharp[SynchronousBreakfast](./snippets/index/AsyncBreakfast-starter/Program.cs#Main)]
+:::code language="csharp" source="snippets/index/AsyncBreakfast-starter/Program.cs" highlight="8-27":::
```

Comment:
Does this need an `access-by-line.txt` file?
Nearly all the samples that are accessed by line number has this file. Example [here](https://github.com/dotnet/docs/blob/master/samples/snippets/csharp/tour/statements/access-by-line.txt).

Reply:
No, this is intentional. I want the entire file rendered on the doc, and then I highlight `Main`. I do this for the starter and final versions.

Reply:
@IEvangelist. Yes I know that you want to render the entire file and then highlight specific lines.
The `access-by-line.txt` won't change that behavior.

See https://github.com/dotnet/samples/pull/1188#issuecomment-523668156 for why `access-by-line.txt` files were created.

Reply:
Ah, ok. I was actually unaware of this. So, it must be used to let our build system know what files are consuming it be line number. That way if someone later changes the source snippet - the build system will re-render those depending pages accordingly - yes. We should add that.

Reply:
I'm not sure if the build system does that automatically now.
When the access-by-line was first added, it didn't have anything done automatically. It just alert when someone modifies a code sample so he don't forget to update line numbers.

Reply:
Addressed in #18633